price_parser/parser.py

```diff
@@ -90,12 +90,11 @@ def or_regex(symbols: List[str]) -> Pattern:
 # can be written as SGD$123 or NZD $123. Currency code should take priority
 # over $ symbol in this case.
 DOLLAR_CODES = [k for k in CURRENCY_CODES if k.endswith('D')]
-DOLLAR_REGEXES = [
-    r"""
-    \b{}   # code like NZD
-    (?:[^\w]|$)  # not a letter
-    """.format(k) for k in DOLLAR_CODES
-]
+_DOLLAR_REGEX = re.compile(
+    r'\b(?:{})(?=\$?)\b'.format(
```

Comment:
Hey @Gallaecio! Do you think it is possible to preserve comments in the regex, to keep it readable?

Also, it seems the regex is different now; If I'm not mistaken, handling of NZD50 could have changed - is it intentional?

Reply:
Actually, because `\w` includes digits, handling of NZD50 did not change. I’ve just added two test cases to cover the scenario you mention, and modified the expression to support them.

However, I’m not sure the new scenarios can be found in real life, and I’m not sure we should include them unless that’s the case.